build.gradle

```diff
@@ -107,23 +134,56 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String findNewestEapBuildNumber() {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?)-EAP-SNAPSHOT.pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    return matcher.group(1)
+  }
+  return null
+}
 
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+String checkForEapOfFutureRelease(String intellijMajorPartOfBuildNumber) {
+  String foundBuildNumber = project.findNewestEapBuildNumber()
+  if (foundBuildNumber) {
+    if (Integer.parseInt(majorPartOfBuildNumber(foundBuildNumber)) > Integer.parseInt(intellijMajorPartOfBuildNumber)) {
```

Comment:
Actually... would `VersionNumber.parse` work here as well?

As in, is would

```
VersionNumber.parse(foundBuildNumber) > VersionNumber.parse(intellijMajorPartOfBuildNumber)
```

be true if `foundBuildNumber == "222.123.456"` and `intellijMajorPartOfBuildNumber == "221"`?

Reply:
Coz if so... then `checkForEapOfFutureRelease` and `checkForEapWithBuildNumberHigherThan` can literally be merged into a single function!

Reply:
Or rather, the current form of `checkForEapWithBuildNumberHigherThan` can be used for both cases (`latestEapBuildNumber` being null or non-null), just with a different param 🤔

Reply:
yes, but when we pass null (in case when `eapOfLatestSupportedMajor` is empty in properties file) I will need to call project.intellijVersion in `checkForEapWithBuildNumberHigherThan ` method, it will be okey ?

Reply:
so I can add additional param to this method or just check in this if `latestEapBuildNumber` is not empty from `project.intellijVersions`